src/main/java/io/leonis/subra/game/data/Goal.java

```diff
@@ -1,56 +1,25 @@
 package io.leonis.subra.game.data;
 
 import io.leonis.algieba.Spatial;
-import io.leonis.algieba.geometry.CardinalDirection;
-import java.io.Serializable;
+import io.leonis.subra.game.data.Team.TeamIdentity;
 import java.util.Set;
 import lombok.Value;
+import lombok.experimental.Delegate;
 import org.nd4j.linalg.api.ndarray.INDArray;
-import org.nd4j.linalg.indexing.NDArrayIndex;
 
 /**
  * The Interface Goal.
  *
- * This interface describes the functionality of a goal.
+ * This interface describes the state of a goal.
  *
- * @author Rimon Oz
  * @author Jeroen de Jong
- * @author Ryan Meulenkamp
  */
-public interface Goal extends Spatial, Serializable {
-  /**
-   * @return The {@link CardinalDirection} to which the back of the goal is facing.
-   */
-  CardinalDirection getCardinalDirection();
-
-  /**
-   * @return The {@link TeamColor color} of the goal.
-   */
-  TeamColor getTeamColor();
+public interface Goal extends GoalDimension, Spatial, FieldHalf.Supplier {
```

Comment:
Does it make sense that a `Goal` extends `GoalDimension` ? It just has the same properties; so maybe `Goal` should implement `GoalDimension.Supplier` ?